estimatey-api/Estimatey.Infrastructure/Float/FloatHistoricLoggedTimeSyncer.cs

```diff
@@ -48,40 +51,50 @@ public async Task SyncHistoricLoggedTime()
 
     private async Task SyncProject(ProjectEntity project, List<FloatPersonEntity> people)
     {
-        var syncLoggedTimeUntil = DateOnly.FromDateTime(_dateTimeService.Now.AddDays(-_persistLoggedTimeBeforeDaysAgo - 1));
+        var semaphore = _semaphores.GetOrAdd(project.Id, _ => new SemaphoreSlim(1, 1));
+        await semaphore.WaitAsync();
 
-        if (project.LoggedTimeHasBeenSyncedUntil is not null &&
-            project.LoggedTimeHasBeenSyncedUntil >= syncLoggedTimeUntil)
+        try
         {
-            _logger.LogInformation("Historic logged time is up to date so no need to sync.");
-            return;
-        }
+            var syncLoggedTimeUntil = DateOnly.FromDateTime(_dateTimeService.Now.AddDays(-_persistLoggedTimeBeforeDaysAgo - 1));
 
-        DateOnly? syncLoggedTimeFrom = project.LoggedTimeHasBeenSyncedUntil.HasValue ?
-            project.LoggedTimeHasBeenSyncedUntil.Value.AddDays(1) :
-            null;
+            if (project.LoggedTimeHasBeenSyncedUntil is not null &&
+                project.LoggedTimeHasBeenSyncedUntil >= syncLoggedTimeUntil)
+            {
+                _logger.LogInformation("Historic logged time is up to date so no need to sync.");
+                return;
+            }
 
-        _logger.LogDebug("Syncing historic logged time for project {projectId} between {startDate} and {endDate}.", project.Id, syncLoggedTimeFrom, syncLoggedTimeUntil);
+            DateOnly? syncLoggedTimeFrom = project.LoggedTimeHasBeenSyncedUntil.HasValue ?
+                project.LoggedTimeHasBeenSyncedUntil.Value.AddDays(1) :
+                null;
 
-        var historicLoggedTime = await _floatClient.GetLoggedTime(project.FloatId, syncLoggedTimeFrom, syncLoggedTimeUntil);
+            _logger.LogDebug("Syncing historic logged time for project {projectId} between {startDate} and {endDate}.", project.Id, syncLoggedTimeFrom, syncLoggedTimeUntil);
 
-        var loggedTimeEntities = historicLoggedTime.Select(_ => new LoggedTimeEntity
-        {
-            FloatId = _.Id,
-            FloatPersonId = people.First(person => person.FloatId == _.PersonId).Id,
-            Date = DateOnly.Parse(_.Date),
-            Hours = _.Hours,
-            Locked = _.Locked,
-            LockedDate = _.LockedDate is null ? null : DateOnly.Parse(_.LockedDate),
-        });
+            var historicLoggedTime = await _floatClient.GetLoggedTime(project.FloatId, syncLoggedTimeFrom, syncLoggedTimeUntil);
 
-        _logger.LogDebug("Persisting {loggedTimeEntriesCount} logged time entries.", loggedTimeEntities.Count());
+            var loggedTimeEntities = historicLoggedTime.Select(_ => new LoggedTimeEntity
```

Comment:
Do you want a `LoggedTimeDto.MapToEntity`?

Reply:
Nah, just going to leave as feels wrong for some reason. No bigge.